src/FastSerialization/SegmentedList.cs

```diff
@@ -328,13 +339,14 @@ public void AppendFrom(SegmentedList<T> from, int index, int count)
 
                 do
                 {
-                    int sourceSegment = index / from.segmentSize;
-                    int sourceOffset = index % from.segmentSize;
+                    int sourceSegment = (int)(index / from.segmentSize);
+                    int sourceOffset = (int)(index % from.segmentSize);
                     int sourceLength = from.segmentSize - sourceOffset;
-                    int targetSegment = this.count >> this.segmentShift;
-                    int targetOffset = this.count & this.offsetMask;
+                    int targetSegment = (int)(this.count >> this.segmentShift);
+                    int targetOffset = (int)(this.count & this.offsetMask);
                     int targetLength = this.segmentSize - targetOffset;
-                    int countToCopy = Math.Min(count, Math.Min(sourceLength, targetLength));
+                    // We can safely cast to int since source and target lengths will never surpass int.MaxValue
+                    int countToCopy = (int)Math.Min(count, Math.Min(sourceLength, targetLength));
```

Comment:
Is this safe to cast, as count is now a long?

Reply:
Yes, it is safe. Because this logic copies stuff over by segments, which are always 8_192 items long.